docs/api_guide/plot_06_sklearn_pipeline_cv_demo.py

```diff
@@ -0,0 +1,473 @@
+"""
+# Selecting basis by cross-validation with scikit-learn
+
+In this demo, we will demonstrate how to select an appropriate basis and its hyperparameters using cross-validation.
+In particular, we will learn:
+
+1. What a scikit-learn pipeline is.
+2. Why pipelines are useful.
+3. How to combine NeMoS `Basis` and `GLM` objects in a pipeline.
+4. How to select the number of bases and the basis type through cross-validation (or any other hyperparameter in the pipeline).
+5. How to use a custom scoring metric to quantify the performance of each configuration.
+
+"""
+
+# %%
+# ## What is a scikit-learn pipeline
+#
+# <figure markdown>
+# <img src="../../../assets/pipeline.svg" style="width: 100%"/>
+# <figcaption>Schematic of a scikit-learn pipeline.</figcaption>
+# </figure>
+#
+# A pipeline is a sequence of data transformations. Each step in the pipeline transforms the input data into a different representation. The final step is either another transformation or a model step that fits, predicts, or scores based on the previous step's output and some observations. Setting up such machinery can be simplified using the `Pipeline` class from scikit-learn.
+#
+# To set up a scikit-learn `Pipeline`, ensure that:
+#
+# 1. Each intermediate step is a [scikit-learn transformer object](https://scikit-learn.org/stable/data_transforms.html) with a `transform` and/or `fit_transform` method.
+# 2. The final step is either another transformer or an [estimator object](https://scikit-learn.org/stable/developers/develop.html#estimators) with a `fit` method, or a model with `fit`, `predict`, and `score` methods.
+#
+# Each transformation step takes a 2D array `X` of shape `(num_samples, num_original_features)` as input and outputs another 2D array of shape `(num_samples, num_transformed_features)`. The final step takes a pair `(X, y)`, where `X` is as before, and `y` is a 1D array of shape `(n_samples,)` containing the observations to be modeled.
+#
+# You can define a pipeline as follows:
+# ```python
+# from sklearn.pipeline import Pipeline
+#
+# # Assume transformer_i/predictor is a transformer/model object
+# pipe = Pipeline(
+#     [
+#         ("label_1", transformer_1),
+#         ("label_2", transformer_2),
+#         ...,
+#         ("label_n", transformer_n),
+#         ("label_model", model)
```

Comment:
should say that these labels are arbitrary and where/how they're used (only for display, right?)